One logger per module, and some associated changes:

julia326 opened the pull request:
- all print statements are now logs
- some infos are now debugs, was too verbose before
- not adding a logging.config yet, may add later if mhctools often used as binary

Fixes #62, #68 

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/hammerlab/mhctools/63)

<!-- Reviewable:end -->

julia326 committed:
One logger per module, and some associated changes:
- all print statements are now logs
- some infos are now debugs, was too verbose before
- not adding a logging.config yet, may add later if mhctools often used as binary

ihodes commented:
LGTM--I'd try rerunning Travis tests tomorrow, looks like IEDB was down?

iskandr commented on mhctools/cli/script.py:
```diff
@@ -57,6 +62,6 @@ def main(args_list=None):
     }
     epitope_collection = predictor.predict(input_dictionary)
     df = epitope_collection.to_dataframe()
-    print(df)
```
Printing this DataFrame is sometimes the only output I care about (e.g. when checking if there are any strong MHC binding subsequences within one particular sequence), so I'd rather leave this as a print (or at least promote the log statement to `info`).

iskandr commented on mhctools/process_helpers.py:
```diff
@@ -108,8 +112,10 @@ def run_multiple_commands_redirect_stdout(
-                  process.redirect_stdout_file.name)
+            handler = logging.FileHandler(process.redirect_stdout_file.name)
+            handler.setLevel(logging.DEBUG)
+            logger.addHandler(handler)
```
It seems like the logger could accumulate many handlers from dead processes (sometimes this code runs tens of thousands of times), do you think that will be a problem?

julia326 replied:
so in that situation, does it currently write to tens of thousands of files? or it runs 20k times but only writes to a handful of files?

julia326 replied:
adding a line that removes the handler after the debug statements get written

julia326 committed:
logger.debug -> info for outputting resulting DF

julia326 committed:
Adding a logging.conf for mhctools script

julia326 committed:
adding logging.conf

julia326 committed:
removing logging.conf oops

julia326 committed:
adding newline for df output

julia326 committed:
Adding a RELEASING.md, upping version number in preparation for increase

julia326 committed:
Moving up one dir

julia326 committed:
some more changes:
- using pkg_resources
- upping dependent versions
- fixing some RELEASING.md numbers/typos

julia326 committed:
removing extra RELEASING.md

julia326 committed:
Some more changes:
- adding RELEASING.md
- removing handlers after debug output is printed (in process_helpers.py) to avoid accumulating lots of handlers
- using pkg_resources for reading in logging.conf
